subprojects/core-api/src/main/java/org/gradle/plugin/use/PluginDependencySpec.java

```diff
@@ -48,6 +51,26 @@
      */
     PluginDependencySpec version(@Nullable String version);
 
+    /**
+     * Specify the version of the plugin to depend on.
+     *
+     * <pre>
+     * plugins {
+     *     id "org.company.myplugin" version libs.versions.myplugin
+     * }
+     * </pre>
+     *
+     * @param version the version provider, for example as found in a version catalog
+     * @return this
+     *
+     * @since 7.2
+     */
+    @Incubating
+    default PluginDependencySpec version(Provider<String> version) {
```

Comment:
If I understand correctly, this is an enhancement of the plugins block `version` part, which can be used without or without the catalog, correct?

I wonder about making the `forUseAtConfigurationTime()` implicit vs. requiring the user to explicitly use it. Is that a pattern used elsewhere?

Reply:
There's no way to use a `Provider` today in the plugins block. With this PR this is possible using the catalog version providers, but that's pretty much only this.

Reply:
Given the limitations of the `plugins` block, this makes sense.